### ISSUE TITLE
Connection to server got closed. Server will not be restarted

### ISSUE THREAD

**Issue by homonto** [microsoft/vscode-cpptools#10444]
### Environment

- OS and Version:  macOS Ventura 13.1 (22C65)
- VS Code Version: 1.74.3
- C/C++ Extension Version: v1.13.9

### Bug Summary and Steps to Reproduce

Bug Summary:

[Error - 9:12:57 AM] Connection to server got closed. Server will not be restarted.
[Error - 9:12:57 AM] Server initialization failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 9:12:57 AM] cpptools client: couldn't create connection to server.
  Message: Pending response rejected since connection got disposed
  Code: -32097 

so the extension is not functional
<img width="676" alt="Screenshot 2023-01-31 at 09 21 45" src="https://user-images.githubusercontent.com/46562447/215720282-55f0e6f7-179d-4a55-85cc-5f5b7f4e0b25.png">
<img width="486" alt="Screenshot 2023-01-31 at 09 21 35" src="https://user-images.githubusercontent.com/46562447/215720286-4543db05-dd0e-4e93-8373-274c8755ec92.png">

### Configuration and Logs

```shell
[Error - 9:24:27 AM] Connection to server got closed. Server will not be restarted.
[Error - 9:24:27 AM] Server initialization failed.
  Message: Pending response rejected since connection got disposed
  Code: -32097 
[Error - 9:24:27 AM] cpptools client: couldn't create connection to server.
  Message: Pending response rejected since connection got disposed
  Code: -32097
```

**Comment by michelleangela** [comment]
@homonto 
Is there an isolated sample code you could provide that reproduces the issue? Are there specific repro steps to cause the extension to crash, such as invoking a specific IntelliSense feature?

**Comment by homonto** [comment]
I am desperate to remove both VSC and Platformio and install everything from zero but on the other hand I am keeping it this way so at least I can serve you with the info if you need to troubleshoot it 
but if you don't need my info I will reinstall everything from the scratch (but that is not .... "my way" - it is a...."windows")

**Comment by KuroshM** [comment]
Exactly the same issue here.

**Comment by michelleangela** [comment]
@homonto 
Could you try the pre-release version 1.14.0 (https://github.com/microsoft/vscode-cpptools/releases/tag/v1.14.0). We fixed some issues in 1.14.0 that could be causing a crash to the extension's language server.

**Comment by michelleangela** [comment]
> Exactly the same issue here.

@KuroshM 
Are you also using the same version of the extension on the same platform? Does it also crash for any code base including workspace folder with a simple source file?

Could you try the pre-release version 1.14.0 (https://github.com/microsoft/vscode-cpptools/releases/tag/v1.14.0). ?

**Comment by homonto** [comment]
> @homonto Could you try the pre-release version 1.14.0 (https://github.com/microsoft/vscode-cpptools/releases/tag/v1.14.0). We fixed some issues in 1.14.0 that could be causing a crash to the extension's language server.

mucios gracias senior - it works ;-)
thank you

<img width="171" alt="Screenshot 2023-02-01 at 20 44 31" src="https://user-images.githubusercontent.com/46562447/216159242-c12e1c6b-9b21-4ba7-8cf3-8ddceeb3eace.png">
<img width="217" alt="Screenshot 2023-02-01 at 20 43 17" src="https://user-images.githubusercontent.com/46562447/216159246-cca17c09-f325-419d-951a-38bb4432aeec.png">
<img width="884" alt="Screenshot 2023-02-01 at 20 42 57" src="https://user-images.githubusercontent.com/46562447/216159248-50586cdb-a4ef-4772-85f0-9f865dcd3808.png">

**Comment by KuroshM** [comment]
> > Exactly the same issue here.
> 
> @KuroshM Are you also using the same version of the extension on the same platform? Does it also crash for any code base including workspace folder with a simple source file?
> 
> Could you try the pre-release version 1.14.0 (https://github.com/microsoft/vscode-cpptools/releases/tag/v1.14.0). ?

Hi,
I am using WSL on windows 10. VS code version 1.74.3, C/C++ extension version v1.13.9. It usually takes some time to crash, I get the notification, however intellisense features work even after I get the notification. Today (using the pre-release version) I have not got the notification so far. I will update if I got it.
Thanks!
